src/Common/src/TypeSystem/RuntimeDetermined/CanonUtilites.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem
```

Comment:
We shouldn't end up with a separate copy of this file for scenarios where we have runtime determined types and when we don't. Either we don't allow the existing model of having canonical things without runtimedetermined things, or we make this logic into an algorithm.